streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -144,6 +144,8 @@ public void close(final boolean clean,
         } finally {
             closeStateManager(committedSuccessfully);
         }
+
+        taskClosed = true;
```

Comment:
I am not sure why we need this? Should the unsubscribe/subscribe change not fix the issue? What is the scenario that is not covered by unsubscribe/subscribe that is fixed with `taskClosed` flag?

Reply:
Strictly speaking this change is not required for this particular fix itself, but I just want to add some more safety net on the code in case there are other unknown issues. After second thought, what I can do is to change the thrown exception to `IllegalStateException` to let it fail immediately than trying to cover any unknown issues. WDYT?

Reply:
Not sure. Both approaches have advantages and disadvantages.

The proposed change make the code more robust and thus might result in better user experience. However, we might not learn about issues as user might never recognize this log entry and thus never report. (Maybe we should log this as an ERROR and explain in the message that Streams will auto-recover from this, but people should please report that this happened? -- This would at least increase the likelihood to get a report.)

For `IllegalStateException` it would be the other way round... Guess it depends on the likelihood of a bug hitting this? If it's rare, failing might be better?

But not sure. \cc @bbejeck @dguy

Reply:
I kept the migration exception while bumping the log to WARN indicating this is not expected to happen in the new commit.